newIDE/app/src/Profile/EditProfileDialog.js

```diff
@@ -27,115 +28,122 @@ type Props = {|
   error: ?AuthError,
 |};
 
-type State = {|
-  form: EditForm,
-|};
-
 export const getUsernameErrorText = (error: ?AuthError) => {
   if (!error) return undefined;
 
   if (error.code === 'auth/username-used')
     return 'This username is already used, please pick another one.';
-  if (error.code === 'auth/malformed-username') return usernameFormatError;
+  if (error.code === 'auth/malformed-username')
+    return usernameFormatErrorMessage;
   return undefined;
 };
 
-export default class EditDialog extends Component<Props, State> {
-  state = {
-    form: {
-      username: this.props.profile.username || '',
-      description: this.props.profile.description || '',
-      getGameStatsEmail: !!this.props.profile.getGameStatsEmail,
-    },
-  };
+const EditDialog = ({
+  profile,
+  onClose,
+  onEdit,
+  editInProgress,
+  error,
+}: Props) => {
+  const [username, setUsername] = React.useState(profile.username || '');
+  const [description, setDescription] = React.useState(
+    profile.description || ''
+  );
+  const [getGameStatsEmail, setGetGameStatsEmail] = React.useState(
+    !!profile.getGameStatsEmail
+  );
+  const [
+    usernameAvailability,
+    setUsernameAvailability,
+  ] = React.useState<?UsernameAvailability>(null);
+  const [
+    isValidatingUsername,
+    setIsValidatingUsername,
+  ] = React.useState<boolean>(false);
 
-  _onEdit = () => {
-    if (!this._canEdit()) return;
+  const canEdit = React.useMemo(
+    () =>
+      !editInProgress &&
```

Comment:
editInProgress is a bit ambiguous here, I thought that it was to say "Is the user editing the description?" or another field. To use the same convention as for the creation, it could be updateProfileInProgress